notify/telegram/telegram.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telegram
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"gopkg.in/telebot.v3"
+
+	"github.com/prometheus/alertmanager/template"
+
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/types"
+	commoncfg "github.com/prometheus/common/config"
+)
+
+// Notifier implements a Notifier for telegram notifications.
+type Notifier struct {
+	conf    *config.TelegramConfig
+	tmpl    *template.Template
+	logger  log.Logger
+	client  *http.Client
+	retrier *notify.Retrier
+}
+
+// New returns a new Telegram notification handler.
+func New(conf *config.TelegramConfig, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*conf.HTTPConfig, "telegram", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+	return &Notifier{
+		conf:    conf,
+		tmpl:    t,
+		logger:  l,
+		client:  client,
+		retrier: &notify.Retrier{},
+	}, nil
+}
+
+func (n *Notifier) Notify(ctx context.Context, alert ...*types.Alert) (bool, error) {
+	var (
+		err  error
+		data = notify.GetTemplateData(ctx, n.tmpl, alert, n.logger)
+		tmpl = notify.TmplText(n.tmpl, data, &err)
+	)
+
+	client, err := n.createTelegramClient()
+	if err != nil {
+		return true, err
+	}
+
+	// Telegram supports 4096 chars, but message still would be too long
+	messageText, truncated := notify.Truncate(tmpl(n.conf.Message), 1024)
+	if truncated {
```

Comment:
This comment is not clear to me. How did you choose the 1024?

Reply:
I can't remember what I was thinking when I wrote this line, changed size to 4096 chars 😅

Reply:
This has caused frustration with the alertmanager-bot telegram integration over and over again. It would be great to somehow be able to split messages into multiple if it needs end up being truncated. 
We didn't implement this so far, but it frequently comes up:
https://github.com/metalmatze/alertmanager-bot/discussions/151

Reply:
Nothing that needs to be added to this PR but surely a good thing to follow up with in the future. :relaxed: